File: 785. Is Graph Bipartite.py
class Solution:
    def isBipartite(self, graph: List[List[int]]) -> bool:
        group, visit, queue = [-1]*len(graph), [0]*len(graph), []
        
        for i in range(len(graph)):
            if not visit[i]:
                queue.append(i)
            
            while queue:
                cur = queue.pop(0)
                visit[cur] = 1
                
                if group[cur]==-1:
                    group[cur] = 0
                
                near = (group[cur]+1)%2
                
                for adj in graph[cur]:
                    if group[adj]!=-1 and group[adj]!=near:
                        return False
                    
                    group[adj] = near
                    
                    if not visit[adj]:
                        queue.append(adj) 
        
        return True

File: 785. Is Graph Bipartite.js
/**
 * @param {number[][]} graph
 * @return {boolean}
 */
var isBipartite = function(graph) {
    let [group, visit, queue] = [new Array(graph.length).fill(-1), new Array(graph.length).fill(0), []];

    for(let i=0;i<graph.length;i++){
        if(!visit[i]){
            queue.push(i);
        }
        while(queue.length>0){
            let cur = queue.shift();
            visit[cur] = 1;

            if(group[cur]==-1){
                group[cur] = 0;
            }
            let near = (group[cur]+1)%2;

            for(adj of graph[cur]){
                if(group[adj]!=-1 && group[adj]!=near){
                    return false;
                }
                group[adj] = near;
                
                if(!visit[adj]){
                    queue.push(adj);
                }
            }
        }
    }
    return true;
};